lib/matplotlib/tri/tripcolor.py

```diff
@@ -120,8 +120,9 @@ def tripcolor(ax, *args, alpha=1.0, norm=None, cmap=None, vmin=None,
             raise ValueError(
                 "shading='gouraud' can only be used when the colors "
                 "are specified at the points, not at the faces.")
-        collection = TriMesh(tri, **kwargs)
-        colors = point_colors
+        _api.check_isinstance((Normalize, None), norm=norm)
```

Comment:
I would move this up before the if (l.118). Then you don't have to duplicate the code for the else branch.